.github/COPILOT_INSTRUCTIONS.md

```diff
@@ -0,0 +1,321 @@
+
+## Overview
+This document provides specific instructions for GitHub Copilot when reviewing pull requests for the CMS MINT application. These guidelines ensure consistent code quality, security, and adherence to established patterns.
+
+## Code Review Priorities
+
+### 🔴 Critical Issues (Request Changes)
+1. **Security vulnerabilities**
+2. **Breaking changes without proper migration**
+3. **Missing authentication/authorization checks**
+4. **Hardcoded secrets or sensitive data**
+5. **SQL injection vulnerabilities**
+6. **XSS vulnerabilities**
+7. **Missing input validation**
+8. **Disabled tests without justification**
+
+### 🟡 Major Issues (Strong Suggestions)
+1. **TypeScript `any` type usage**
+2. **Console statements in production code**
+3. **TODO/FIXME comments without tracking**
+4. **ESLint disable comments without justification**
+5. **Non-strict equality operators (`==`, `!=`)**
+6. **Missing error handling**
+7. **Performance issues (N+1 queries, inefficient loops)**
+8. **Accessibility violations**
+
+### 🟢 Minor Issues (Suggestions)
+1. **Naming convention violations**
+2. **Code formatting inconsistencies**
+3. **Missing documentation**
+4. **Overly complex functions**
+5. **Unused imports or variables**
+
+## Technology-Specific Guidelines
+
+### Go Backend Review
+
+#### Required Patterns to Check
+- **Base Struct Pattern**: All models should extend `baseStruct` with proper UUID ID
+**Store Pattern**: CRUD operations should follow `func (s *Store) {EntityName}Create(np sqlutils.NamedPreparer, logger *zap.Logger, {entity} *models.{EntityName}) (*models.{EntityName}, error)` pattern (e.g., `ModelPlanCreate`, `UserAccountCreate`)
+- **Handler Pattern**: All handlers should extend `HandlerBase` and use structured error responses
+- **Principal Pattern**: Authentication checks should use `Principal` interface
+- **Logging Pattern**: Use structured logging with Zap, include context and trace IDs
+
+#### Go Anti-patterns to Flag
+```go
+// ❌ Direct database queries without transaction support
+db.Query("SELECT * FROM users WHERE id = ?", id)
+
+// ❌ Missing error handling
+result, _ := someFunction()
+
+// ❌ Hardcoded strings instead of constants
+if role == "admin" {
+
+// ❌ Bare error passthrough (acceptable currently, but prefer wrapped errors)
+if err != nil {
+    return nil, err  // Consider: errors.Wrap(err, "context about operation")
+}
+
+// ❌ Non-descriptive function names
+func DoStuff() {}
+
+// ❌ Missing authorization checks
+func SensitiveOperation() {}
+```
+
+#### Go Best Practices to Enforce
+```go
+// ✅ Proper transaction handling
+func (s *Store) {EntityName}Create(np sqlutils.NamedPreparer, logger *zap.Logger, {entity} *models.{EntityName}) (*models.{EntityName}, error) {
+    // Implementation
+}
+
+// ✅ Preferred: Error wrapping with context (moving toward this pattern)
+if err != nil {
+    logger.Error("Failed to create model", zap.Error(err), zap.String("operation", "ModelCreate"))
```

Comment:
We don't always explicitly log the error here either, only if it is a case that we explicitly want a splunk alert. I'm not sure if we would want to wrap the operation here either, we get some of this for free just with the logger. I'm not sure it hurts to include it though either.